elytron/WFCORE-5178_native-support-for-oidc.adoc

```diff
@@ -0,0 +1,178 @@
+OIDC HTTP mechanism implementation for the deployment. The deployment will make use of a "Virtual Security Domain",
+similar to what was implemented for https://github.com/wildfly/wildfly-proposals/blob/master/microprofile/WFLY-12609_MicroProfile_JWT_1_1.adoc#hard-requirements[Microprofile JWT].
+This means that it will no longer be necessary to pre-configure SecurityDomains in the Elytron subsystem,
+as was required with the Keycloak client adapter.
+
+It should be possible to make use of different OpenID providers with WildFly. While working on the
+implementation for this RFE, we should make sure that we aren't including Keycloak-specific configuration, if
+possible.
+
+==== New galleon layer for OIDC support
+
+A new Galleon layer, `oidc`, should be added to allow a server to be provisioned with OIDC support.
+
+=== Nice-to-Have Requirements
+
+* https://openid.net/specs/openid-connect-rpinitiated-1_0.html[RP-Initiated Logout] is an optional
+OIDC specification that defines how a relying party (RP) can request that an OpenID provider logs
+out the end-user. The Keycloak client adapter provides support for this so this is something we
+should provide as well but we could possibly look at that as a follow-up to this RFE if necessary.
+
+* The existing Keycloak adapter can also be used to secure the admin console. This requires some
+additional work so it might be done as a follow-up task.
+
+* *Migration*: For the most part, the same subsystem attributes and JSON configuration options that could
+be used with the `keycloak-adapter-subsystem` and the `keycloak.json` file will still work with the
+`elytron-oidc-subsystem` and the `oidc.json` file. The main difference is the actual subsystem name and
+the name of the JSON configuration file. That means that the migration tool could potentially detect an
+existing `keycloak` subsystem configuration and then change its name to the new subsystem
+name instead (and update any attributes/elements if needed). For the deployment side, the migration tool
+could detect deployments with `auth-method` set to KEYCLOAK and with a keycloak.json configuration file and
+update the `auth-method` to OIDC and rename the configuration file to oidc.json.
+
+=== Non-Requirements
+
+This integration will not be supporting the use of pre-configured SecurityDomains. The purpose of OIDC
+is to allow a client to verify the identity of an end-user based on authentication that is performed
+by an authorization server. Information about the end-user can be obtained from ID tokens. Scope
+information (i.e., role information) can be obtained from access tokens. Thus, we do not need
+pre-configured security resources in the Elytron subsystem.
+
+Support for the following optional OIDC specifications are currently non-requirements. These
+are also not provided by the Keycloak Java client adapter:
+
+* https://openid.net/specs/openid-connect-discovery-1_0.html[Discovery]
```

Comment:
Is this (partially?) provided and leveraged when using `provider-url`?

Reply:
The discovery specification is actually something that an OpenID provider implements. I'll remove it from the list here since our implementation is concerned with the client side (i.e., relying party config). 

In general though, the OpenID Connect spec mentions that metadata about an OpenID provider is usually obtained via discovery. The `provider-url` is a URL for an OpenID provider that can be used to discover information about it via the `provider-url/.well-known/openid-configuration` url.